### ISSUE TITLE
Cannot invoke "java.util.Collection.size()" because "storageNodes" is null

### ISSUE THREAD

**Issue by a601942905git** [apache/shardingsphere#29509]
## Question

The following error occurs when shardingsphere integrates consul：Cannot invoke "java.util.Collection.size()" because "storageNodes" is null
config content：
```yam
mode:
  type: Cluster
  repository:
    type: consul
    props:
      namespace: governance_ds
      server-lists: localhost
      timeToLiveSeconds: 30
      blockQueryTimeToSeconds: 60
```
shardingsphere-jdbc-core version is：5.4.1
shardingsphere-cluster-mode-repository-consul version is：5.4.1

**Comment by RaigorJiang** [comment]
Hi @a601942905git 
Do you have configuration in consul, and how to reproduce this exception?

**Comment by a601942905git** [comment]
Consul does not have any configuration. It can be started normally if it is replaced with zookeeper without adding configuration.

**Comment by a601942905git** [comment]
What configuration do I need to add on consul

**Comment by a601942905git** [comment]
> Hi @a601942905git Do you have configuration in consul, and how to reproduce this exception?

Consul does not have any configuration. It can be started normally if it is replaced with zookeeper without adding configuration.

**Comment by a601942905git** [comment]
@RaigorJiang you can see this method：org.apache.shardingsphere.mode.spi.PersistRepository#getChildrenKeys

**Comment by a601942905git** [comment]
@RaigorJiang you can see this method：org.apache.shardingsphere.mode.spi.PersistRepository#getChildrenKeys
zookeeper impl return empty collection
consul impl return null

**Comment by RaigorJiang** [comment]
> @RaigorJiang you can see this method：org.apache.shardingsphere.mode.spi.PersistRepository#getChildrenKeys zookeeper impl return empty collection consul impl return null

Hi @a601942905git , do you mean `response.getValue()` will be null?

<img width="846" alt="image" src="https://github.com/apache/shardingsphere/assets/5668787/4eaf449a-7602-4cf1-8055-3cf45ea50254">

**Comment by a601942905git** [comment]
> > @RaigorJiang you can see this method：org.apache.shardingsphere.mode.spi.PersistRepository#getChildrenKeys zookeeper impl return empty collection consul impl return null
> 
> Hi @a601942905git , do you mean `response.getValue()` will be null?
> 
> <img alt="image" width="846" src="https://private-user-images.githubusercontent.com/5668787/292765049-4eaf449a-7602-4cf1-8055-3cf45ea50254.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM1Njk1NjcsIm5iZiI6MTcwMzU2OTI2NywicGF0aCI6Ii81NjY4Nzg3LzI5Mjc2NTA0OS00ZWFmNDQ5YS03NjAyLTRjZjEtODA1NS0zY2Y0NWVhNTAyNTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjZUMDU0MTA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjYwZjAxMTVkMmY5Njg3ZDg0NGE2ZWUxNjFmNGExMWRmMWY5MDI0OWU1MmNlMDU5MmY5NzJkM2UyNjVkOTk0MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.kaiEgiOASSCySV7ePtVQ2vtFfgA2c7Uo-u0IQJgk4Bc">

yes

**Comment by RaigorJiang** [comment]
> > Hi @a601942905git , do you mean `response.getValue()` will be null?
> yes

In this case, we can judge `response.getValue()` first.  Are you interested in submitting a PR?

**Comment by a601942905git** [comment]
> > > Hi @a601942905git , do you mean `response.getValue()` will be null?
> > > yes
> 
> In this case, we can judge `response.getValue()` first. Are you interested in submitting a PR?

no

**Comment by linghengqian** [comment]
- I also addressed the current issue in #29621. 
- However, the current issue does not involve the problem that `ConsulRepository.watch()` cannot update the local data source. If you are interested, you are welcome to open a new PR or a new issue.

**Comment by linghengqian** [comment]
- Call the original implementer of the Consul integration @gavenpeng . Do you have time to deal with issues related to `ConsulRepository.watch()` failure?
